catalog/app/containers/Bucket/File.js

```diff
@@ -382,15 +382,20 @@ export default function File({
     }),
   })
 
-  const downloadable =
-    !noDownload &&
-    versionExistsData.case({
-      _: () => false,
-      Ok: requests.ObjectExistence.case({
-        _: () => false,
-        Exists: ({ deleted, archived }) => !deleted && !archived,
+  const { downloadable, hash } = versionExistsData.case({
+    _: () => ({
+      downloadable: false,
+    }),
+    Ok: requests.ObjectExistence.case({
+      _: () => ({
+        downloadable: false,
       }),
-    })
+      Exists: ({ deleted, archived, version: versionHash }) => ({
+        downloadable: !noDownload && !deleted && !archived,
+        hash: versionHash,
```

Comment:
i wouldnt call this hash -- it's a version id, it's not linked to the contents of the file and isn't related to any hash functions

Reply:
Ok, got it.
`hash` for package version is valid name, because we use hash function
`hash` for file version is invalid name, because it's some S3 ID

Reply:
I'm renaming this `hash` to `fileVersionId`